### ISSUE TITLE
Use KMPAuth for login

### ISSUE THREAD

**Issue by enthuan** [paug/AndroidMakersApp#189]
I found this library and I think it's a good candidate to implement both Google Sign In and Apple Sign In.

BONUS : Github sign in :)

https://github.com/mirzemehdi/KMPAuth?tab=readme-ov-file

**Comment by martinbonnin** [comment]
Would be awesome indeed. Passkeys would also be nice. Only thing is it binds us to Firebase auth on the backend, which I'm not 100% how to implement. I'll dig more!

**Comment by enthuan** [comment]
Yeah that was the only pain point here.To my opinion KMPauth could be a quick win to implement the current login system on iOS without dealing too much with Apple Sign in.Should I wait before doing this? I have the feeling that passkeys are more difficult to implement, but I may be wrong.Antoine RobiezLe 8 mars 2024 à 11:50, Martin Bonnin ***@***.***> a écrit :﻿
Would be awesome indeed. Passkeys would also be nice. Only thing is it binds us to Firebase auth on the backend, which I'm not 100% how to implement. I'll dig more!

**Comment by martinbonnin** [comment]
Nevermind, I was wrong, we already use [Firebase auth](https://github.com/joreilly/Confetti/blob/5e792f45f1975bb640ff02ebb7f5248c5d959307/backend/service-graphql/src/main/kotlin/dev/johnoreilly/confetti/backend/firebase.kt#L26) in the backend. It should work out of the box with Apple/GitHub/other as long as a firebase idToken is passed in the `Authentication` header. Go for it 👍

**Comment by enthuan** [comment]
It's not that easy to implement. KMPAuth doesn't work the same way as the Firebase unofficial lib.
We can retrieve the 'stored' user with KMPAuth, it only manages (AFAIK) the flow of authentication and return a user once.

**Comment by martinbonnin** [comment]
We _could_ probably store the user locally or so... I was talking signin with @BoD yesterday and would be nice to have to sync bookmarks with the watch. Will dig a bit more.
